lib/Backend/BailOut.cpp

```diff
@@ -585,7 +585,7 @@ BailOutRecord::RestoreValues(IR::BailOutKind bailOutKind, Js::JavascriptCallStac
         }
 #endif
 
-        if (scriptContext->IsInDebugMode())
+        if (functionBody->GetUtf8SourceInfo()->IsUtf8SourceInfoInDebugMode())
```

Comment:
Right now it seems fine as you want clarity on your refactoring. But it will get over-explicit in the future. We could leave the ScriptContextInDebugMode explicitly rest can be converted to IsInDebugMode as they are essentially redirection to the utf8sourceinfo.